src/main/java/com/hazelcast/jdbc/HazelcastJdbcPreparedStatement.java

```
@@ -0,0 +1,370 @@
+        checkClosed();
+    }
+
+    @Override
+    public ParameterMetaData getParameterMetaData() throws SQLException {
+        checkClosed();
+        return null;
+    }
+
+    @Override
+    public void setRowId(int parameterIndex, RowId x) throws SQLException {
+        checkClosed();
+        throw JdbcUtils.unsupported("RowId is not supported");
+    }
+
+    @Override
+    public void setNString(int parameterIndex, String value) throws SQLException {
+        checkClosed();
+        setParameter(parameterIndex, value);
+    }
+
+    @Override
+    public void setNCharacterStream(int parameterIndex, Reader value, long length) throws SQLException {
+        checkClosed();
+        throw JdbcUtils.unsupported("NCharacter Stream is not supported");
+    }
+
+    @Override
+    public void setNClob(int parameterIndex, NClob value) throws SQLException {
+        checkClosed();
+        throw JdbcUtils.unsupported("NClob Stream is not supported");
+    }
+
+    @Override
+    public void setClob(int parameterIndex, Reader reader, long length) throws SQLException {
+        checkClosed();
+        throw JdbcUtils.unsupported("Clob Stream is not supported");
+    }
+
+    @Override
+    public void setBlob(int parameterIndex, InputStream inputStream, long length) throws SQLException {
+        checkClosed();
+        throw JdbcUtils.unsupported("Blob Stream is not supported");
+    }
+
+    @Override
+    public void setNClob(int parameterIndex, Reader reader, long length) throws SQLException {
+        checkClosed();
+        throw JdbcUtils.unsupported("NClob Stream is not supported");
+    }
+
+    @Override
+    public void setSQLXML(int parameterIndex, SQLXML xmlObject) throws SQLException {
+        checkClosed();
+        throw JdbcUtils.unsupported("SQLXML Stream is not supported");
+    }
+
+    @Override
+    public void setObject(int parameterIndex, Object x, int targetSqlType, int scaleOrLength) throws SQLException {
```

It seems that the proper implementation should perform conversion as per the target type. We should either understand in detail how to implement this method properly or do this in a separate issue. In the latter case, it is better to throw an unsupported exception here. The same is applicable to the `setObject(int parameterIndex, Object x, int targetSqlType)` method.

---

Note that these two methods have the default implementations, that throw an unsupported exception. Probably we should just remove our implementations and rely on the defaults.

---

The default implementation is there for the `setObject` overloaded method, so we cannot simply remove this implementation, so it's better to throw an exception for now